toolsrc/include/vcpkg/portfileprovider.h

```diff
@@ -36,4 +37,51 @@ namespace vcpkg::PortFileProvider
         std::vector<fs::path> overlay_ports;
         mutable std::unordered_map<std::string, SourceControlFileLocation> cache;
     };
+
+    struct IVersionedPortfileProvider
+    {
+        virtual const std::vector<vcpkg::Versions::VersionSpec>& get_port_versions(
+            const std::string& port_spec) const = 0;
```

Comment:
```suggestion
            StringView port_spec) const = 0;
```

best to change the rest as well.